src/components/SettingsBlock/SettingsBlock.test.js

```diff
@@ -0,0 +1,101 @@
+import SettingsBlock from '.';
+import {
+  fireEvent,
+  render,
+  screen,
+} from '../../utils/tests/renderWithRouter';
+
+const onClick = jest.fn();
+
+const rendered = (
+  isActive = false,
+  isBoolean = false,
+  icon = 'user',
+  children = 'Settings'
+) =>
+  render(
+    <SettingsBlock
+      onClick={onClick}
+      isActive={isActive}
+      isBoolean={isBoolean}
+      icon={icon}
+    >
+      {children}
+    </SettingsBlock>
+  );
+describe('SettingsBlock Component test suits', () => {
+  it('Should render correctly with its default values', () => {
+    rendered();
+    expect(screen.getByText('Settings')).toBeInTheDocument();
+    expect(screen.getByText('Settings').parentElement).toHaveClass(
+      'group w-full flex items-center justify-between gap-[25px] p-3 bg-white hover:bg-primary-200 cursor-pointer text-gray-500 text-xl font-semibold rounded-lg'
+    );
+  });
+  it('Should render with props correctly', () => {
+    rendered(false, false, 'alert', 'Notifications');
+    expect(screen.getByText('Notifications')).toBeInTheDocument();
+    expect(screen.getByText('Notifications').parentElement).toHaveClass(
+      'group w-full flex items-center justify-between gap-[25px] p-3 bg-white hover:bg-primary-200 cursor-pointer text-gray-500 text-xl font-semibold rounded-lg'
+    );
+    expect(
+      screen.getByText('Notifications').parentElement.firstChild.childNodes[0]
```

Comment:
```suggestion
      screen.getByRole("img")
```

Reply:
There are two img tag in this component, one for `icon` component, other for right chevron (Except when is a toggle component inside)